src/Components/Components/src/EventHandlerInvokerFactory.cs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components
+{
+    public sealed class EventHandlerInvokerFactory
```

Comment:
Still very much a work in progress, but this should give you the gist of it. 

This is an instance (not a static) so it can be targeted by extension methods.